src/Response/ResponseExtensions.cs

```diff
@@ -58,5 +63,59 @@ public static class ResponseExtensions
 
             await negotiator.Handle(response.HttpContext.Request, response, obj, cancellationToken);
         }
+
+        /// <summary>
+        /// Copy a stream into the response body
+        /// </summary>
+        /// <param name="response">Current <see cref="HttpResponse"/></param>
+        /// <param name="stream">The <see cref="Stream"/> to copy from</param>
+        /// <param name="contentType">The content type for the response</param>
+        /// <param name="contentDisposition">The content disposition to allow file downloads</param>
+        /// <returns><<see cref="Task"/>/returns>
+        public static async Task FromStream(this HttpResponse response, Stream source, string contentType, ContentDisposition contentDisposition = null)
+        {
+            long? rangeStart;
+            long? rangeEnd;
+            var contentLength = source.Length;
+
+            response.Headers["Accept-Ranges"] = "bytes";
+            response.ContentType = contentType;
+
+            if (contentDisposition != null)
+            {
+                response.Headers["Content-Disposition"] = contentDisposition.ToString();
+            }
+
+            // rangeHeader should be of the format "bytes=0-" or "bytes=0-12345" or "bytes=123-456"
+            var rangeHeader = response.HttpContext.Request.GetTypedHeaders().Range;
+
+            if (rangeHeader != null)
+            {
+                //Server should return multipart/byteranges; if asking for more than one range but pfft...
```

Comment:
Ain't nobody got time for that!

Reply:
I feel like it should at least throw not supported. Would it not just be as easy as a little `foreach(var range in rangeHeader.Ranges)` ?

Reply:
Nope, looks hard. but...
```csharp
if (rangeHeader.Ranges.Count != 1)
{
    throw something;
}
```
could prevent a surprise.

Reply:
if multiple are parsed then they will get the first.